### DISCUSSION TITLE
TLM Checksum to verify messages received by COSMOS

### DISCUSSION THREAD

**Discussion by shuq0** [OpenSatKit/OpenSatKit#90] [Q&A] [answered]
Hi, 
Is there any reference or sample to append a checksum similar to the CI checksum for every TLM packet being sent out? The crc_checksum.rb file in COSMOS has support for read and write checksum checking. 

Is this feature in CFS already or does that need to be implemented?

**Comment by dmccomas** [comment, marked as answer]
Sorry for the delay and I think I may have seen you post this on the COSMOS
discussions.  The cFS does not directly support this but the CCSDS
secondary header is allowed to be customized. With that said the cFS des
assume telemetry secondary headers to only contain a time stamp.  You could
create a convention for your mission where the first payload byte(s) serve
as a checksum.

- Dave
